backend/main.py

```
@@ -242,0 +1,380 @@
+instrumentator.instrument(app)
+
+GRAFANA_CLOUD_ENDPOINT = os.environ.get("GRAFANA_CLOUD_ENDPOINT")
+GRAFANA_CLOUD_API_KEY = os.environ.get("GRAFANA_CLOUD_API_KEY")
+PROMETHEUS_PUSH_GATEWAY = GRAFANA_CLOUD_ENDPOINT  # Replace with your Grafana Cloud Prometheus remote write endpoint
+
+@app.get("/metrics")
+async def metrics():
+    return instrumentator.expose()
+
+# Push metrics to Grafana Cloud
+@app.on_event("shutdown")
```

Flake8 (7.2.0)

[error] expected 2 blank lines, found 1

(E302)